File: MiniDOMBase/DMD/Anagrafica/Tasks/TaskLavorazioneCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica

    <Serializable>
    Public Class TaskLavorazioneCursor
        Inherits DBObjectCursorPO(Of TaskLavorazione)

        Private m_IDCliente As New CCursorField(Of Integer)("IDCliente")
        Private m_NomeCliente As New CCursorFieldObj(Of String)("NomeCliente")

        Private m_IDStatoAttuale As New CCursorField(Of Integer)("IDStatoAttuale")

        Private m_DataAssegnazione As New CCursorField(Of Date)("DataAssegnazione")
        Private m_IDAssegnatoDa As New CCursorField(Of Integer)("IDAssegnatoDa")

        Private m_DataPrevista As New CCursorField(Of Date)("DataPrevista")
        Private m_IDAssegnatoA As New CCursorField(Of Integer)("IDAssegnatoA")
        Private m_Note As New CCursorFieldObj(Of String)("Note")

        Private m_IDAzioneEseguita As New CCursorField(Of Integer)("IDAzioneEseguita")

        Private m_Categoria As New CCursorFieldObj(Of String)("Categoria")
        Private m_Priorita As New CCursorField(Of Integer)("Priorita")
        Private m_IDSorgente As New CCursorField(Of Integer)("IDSorgente")
        Private m_TipoSorgente As New CCursorFieldObj(Of String)("TipoSorgente")
        Private m_IDContesto As New CCursorField(Of Integer)("IDContesto")
        Private m_TipoContesto As New CCursorFieldObj(Of String)("TipoContesto")

        Private m_IDRegolaEseguita As New CCursorField(Of Integer)("IDRegolaEseguita")
        Private m_ParametriAzione As New CCursorFieldObj(Of String)("ParametriAzione")
        Private m_RisultatoAzione As New CCursorFieldObj(Of String)("RisultatoAzione")
        Private m_DataInizioEsecuzione As New CCursorField(Of Date)("DataEsecuzione")
        Private m_DataFineEsecuzione As New CCursorField(Of Date)("DataFineEsecuzione")
        Private m_IDTaskDestinazione As New CCursorField(Of Integer)("IDTaskDestinazione")
        Private m_IDTaskSorgente As New CCursorField(Of Integer)("IDTaskSorgente")

        Public Sub New()
        End Sub

        Public ReadOnly Property IDRegolaEseguita As CCursorField(Of Integer)
            Get
                Return Me.m_IDRegolaEseguita
            End Get
        End Property

        Public ReadOnly Property ParametriAzione As CCursorFieldObj(Of String)
            Get
                Return Me.m_ParametriAzione
            End Get
        End Property

        Public ReadOnly Property RisultatoAzione As CCursorFieldObj(Of String)
            Get
                Return Me.m_RisultatoAzione
            End Get
        End Property

        Public ReadOnly Property DataInizioEsecuzione As CCursorField(Of Date)
            Get
                Return Me.m_DataInizioEsecuzione
            End Get
        End Property

        Public ReadOnly Property DataFineEsecuzione As CCursorField(Of Date)
            Get
                Return Me.m_DataFineEsecuzione
            End Get
        End Property

        Public ReadOnly Property IDTaskDestinazione As CCursorField(Of Integer)
            Get
                Return Me.m_IDTaskDestinazione
            End Get
        End Property

        Public ReadOnly Property IDTaskSorgente As CCursorField(Of Integer)
            Get
                Return Me.m_IDTaskSorgente
            End Get
        End Property



        Public ReadOnly Property Categoria As CCursorFieldObj(Of String)
            Get
                Return Me.m_Categoria
            End Get
        End Property

        Public ReadOnly Property Priorita As CCursorField(Of Integer)
            Get
                Return Me.m_Priorita
            End Get
        End Property

        Public ReadOnly Property IDSorgente As CCursorField(Of Integer)
            Get
                Return Me.m_IDSorgente
            End Get
        End Property

        Public ReadOnly Property TipoSorgente As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoSorgente
            End Get
        End Property

        Public ReadOnly Property IDContesto As CCursorField(Of Integer)
            Get
                Return Me.m_IDContesto
            End Get
        End Property

        Public ReadOnly Property TipoContesto As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoContesto
            End Get
        End Property

        Public ReadOnly Property IDCliente As CCursorField(Of Integer)
            Get
                Return Me.m_IDCliente
            End Get
        End Property

        Public ReadOnly Property NomeCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCliente
            End Get
        End Property

        Public ReadOnly Property IDStatoAttuale As CCursorField(Of Integer)
            Get
                Return Me.m_IDStatoAttuale
            End Get
        End Property

        Public ReadOnly Property DataAssegnazione As CCursorField(Of Date)
            Get
                Return Me.m_DataAssegnazione
            End Get
        End Property

        Public ReadOnly Property IDAssegnatoDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDAssegnatoDa
            End Get
        End Property

        Public ReadOnly Property DataPrevista As CCursorField(Of Date)
            Get
                Return Me.m_DataPrevista
            End Get
        End Property

        Public ReadOnly Property IDAssegnatoA As CCursorField(Of Integer)
            Get
                Return Me.m_IDAssegnatoA
            End Get
        End Property

        Public ReadOnly Property Note As CCursorFieldObj(Of String)
            Get
                Return Me.m_Note
            End Get
        End Property

        Public ReadOnly Property IDAzioneEseguita As CCursorField(Of Integer)
            Get
                Return Me.m_IDAzioneEseguita
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_TaskLavorazione"
        End Function


        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return Anagrafica.TasksDiLavorazione.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Anagrafica.TasksDiLavorazione.Module
        End Function
    End Class


End Class

File: MiniDOMBase/DMD/Anagrafica/Tasks/TaskLavorazioneCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom.repositories;
using static minidom.Anagrafica;
using static minidom.Sistema;


namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella dei <see cref="TaskLavorazione"/>
        /// </summary>
        [Serializable]
        public class TaskLavorazioneCursor
            : minidom.Databases.DBObjectCursorPO<TaskLavorazione>
        {
            private DBCursorField<int> m_IDCliente = new DBCursorField<int>("IDCliente");
            private DBCursorStringField m_NomeCliente = new DBCursorStringField("NomeCliente");
            private DBCursorField<int> m_IDStatoAttuale = new DBCursorField<int>("IDStatoAttuale");
            private DBCursorField<DateTime> m_DataAssegnazione = new DBCursorField<DateTime>("DataAssegnazione");
            private DBCursorField<int> m_IDAssegnatoDa = new DBCursorField<int>("IDAssegnatoDa");
            private DBCursorField<DateTime> m_DataPrevista = new DBCursorField<DateTime>("DataPrevista");
            private DBCursorField<int> m_IDAssegnatoA = new DBCursorField<int>("IDAssegnatoA");
            private DBCursorStringField m_Note = new DBCursorStringField("Note");
            private DBCursorField<int> m_IDAzioneEseguita = new DBCursorField<int>("IDAzioneEseguita");
            private DBCursorStringField m_Categoria = new DBCursorStringField("Categoria");
            private DBCursorField<int> m_Priorita = new DBCursorField<int>("Priorita");
            private DBCursorField<int> m_IDSorgente = new DBCursorField<int>("IDSorgente");
            private DBCursorStringField m_TipoSorgente = new DBCursorStringField("TipoSorgente");
            private DBCursorField<int> m_IDContesto = new DBCursorField<int>("IDContesto");
            private DBCursorStringField m_TipoContesto = new DBCursorStringField("TipoContesto");
            private DBCursorField<int> m_IDRegolaEseguita = new DBCursorField<int>("IDRegolaEseguita");
            private DBCursorStringField m_ParametriAzione = new DBCursorStringField("ParametriAzione");
            private DBCursorStringField m_RisultatoAzione = new DBCursorStringField("RisultatoAzione");
            private DBCursorField<DateTime> m_DataInizioEsecuzione = new DBCursorField<DateTime>("DataEsecuzione");
            private DBCursorField<DateTime> m_DataFineEsecuzione = new DBCursorField<DateTime>("DataFineEsecuzione");
            private DBCursorField<int> m_IDTaskDestinazione = new DBCursorField<int>("IDTaskDestinazione");
            private DBCursorField<int> m_IDTaskSorgente = new DBCursorField<int>("IDTaskSorgente");

            /// <summary>
            /// Costruttore
            /// </summary>
            public TaskLavorazioneCursor()
            {
            }

            /// <summary>
            /// IDRegolaEseguita
            /// </summary>
            public DBCursorField<int> IDRegolaEseguita
            {
                get
                {
                    return m_IDRegolaEseguita;
                }
            }

            /// <summary>
            /// ParametriAzione
            /// </summary>
            public DBCursorStringField ParametriAzione
            {
                get
                {
                    return m_ParametriAzione;
                }
            }

            /// <summary>
            /// RisultatoAzione
            /// </summary>
            public DBCursorStringField RisultatoAzione
            {
                get
                {
                    return m_RisultatoAzione;
                }
            }

            /// <summary>
            /// DataInizioEsecuzione
            /// </summary>
            public DBCursorField<DateTime> DataInizioEsecuzione
            {
                get
                {
                    return m_DataInizioEsecuzione;
                }
            }

            /// <summary>
            /// DataFineEsecuzione
            /// </summary>
            public DBCursorField<DateTime> DataFineEsecuzione
            {
                get
                {
                    return m_DataFineEsecuzione;
                }
            }

            /// <summary>
            /// IDTaskDestinazione
            /// </summary>
            public DBCursorField<int> IDTaskDestinazione
            {
                get
                {
                    return m_IDTaskDestinazione;
                }
            }

            /// <summary>
            /// IDTaskSorgente
            /// </summary>
            public DBCursorField<int> IDTaskSorgente
            {
                get
                {
                    return m_IDTaskSorgente;
                }
            }

            /// <summary>
            /// Categoria
            /// </summary>
            public DBCursorStringField Categoria
            {
                get
                {
                    return m_Categoria;
                }
            }

            /// <summary>
            /// Priorita
            /// </summary>
            public DBCursorField<int> Priorita
            {
                get
                {
                    return m_Priorita;
                }
            }

            /// <summary>
            /// IDSorgente
            /// </summary>
            public DBCursorField<int> IDSorgente
            {
                get
                {
                    return m_IDSorgente;
                }
            }

            /// <summary>
            /// TipoSorgente
            /// </summary>
            public DBCursorStringField TipoSorgente
            {
                get
                {
                    return m_TipoSorgente;
                }
            }

            /// <summary>
            /// IDContesto
            /// </summary>
            public DBCursorField<int> IDContesto
            {
                get
                {
                    return m_IDContesto;
                }
            }

            /// <summary>
            /// TipoContesto
            /// </summary>
            public DBCursorStringField TipoContesto
            {
                get
                {
                    return m_TipoContesto;
                }
            }

            /// <summary>
            /// IDCliente
            /// </summary>
            public DBCursorField<int> IDCliente
            {
                get
                {
                    return m_IDCliente;
                }
            }

            /// <summary>
            /// NomeCliente
            /// </summary>
            public DBCursorStringField NomeCliente
            {
                get
                {
                    return m_NomeCliente;
                }
            }

            /// <summary>
            /// IDStatoAttuale
            /// </summary>
            public DBCursorField<int> IDStatoAttuale
            {
                get
                {
                    return m_IDStatoAttuale;
                }
            }

            /// <summary>
            /// DataAssegnazione
            /// </summary>
            public DBCursorField<DateTime> DataAssegnazione
            {
                get
                {
                    return m_DataAssegnazione;
                }
            }

            /// <summary>
            /// IDAssegnatoDa
            /// </summary>
            public DBCursorField<int> IDAssegnatoDa
            {
                get
                {
                    return m_IDAssegnatoDa;
                }
            }

            /// <summary>
            /// DataPrevista
            /// </summary>
            public DBCursorField<DateTime> DataPrevista
            {
                get
                {
                    return m_DataPrevista;
                }
            }

            /// <summary>
            /// IDAssegnatoA
            /// </summary>
            public DBCursorField<int> IDAssegnatoA
            {
                get
                {
                    return m_IDAssegnatoA;
                }
            }

            /// <summary>
            /// Note
            /// </summary>
            public DBCursorStringField Note
            {
                get
                {
                    return m_Note;
                }
            }

            /// <summary>
            /// IDAzioneEseguita
            /// </summary>
            public DBCursorField<int> IDAzioneEseguita
            {
                get
                {
                    return m_IDAzioneEseguita;
                }
            }

            //public override string GetTableName()
            //{
            //    return "tbl_TaskLavorazione";
            //}

            //protected internal override CDBConnection GetConnection()
            //{
            //    return TasksDiLavorazione.Database;
            //}

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.TasksDiLavorazione; //.Module;
            }
        }
    }
}